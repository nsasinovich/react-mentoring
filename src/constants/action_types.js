export default {
    CHANGE_SORT: 'CHANGE_SORT',
    UPDATE_SEARCH_INPUT: 'UPDATE_SEARCH_INPUT',
};
