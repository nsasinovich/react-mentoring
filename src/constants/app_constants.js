export const PosterSizes = {
    LARGE: 'poster-large',
    MEDIUM: 'poster-medium',
};

export const FilterOptions = {
    GENRE: 'genres',
    TITLE: 'title',
};

export const SortingOptions = {
    RELEASE: {
        name: 'release date',
        sortField: 'year',
    },
    RATING: {
        name: 'rating',
        sortField: 'vote_count',
    },
};

export const INITIAL_STATE = {
    results: [],
    resultsCount: 0,
    searchInput: '',
    selectedSort: SortingOptions.RATING,
    selectedFilter: FilterOptions.TITLE,
    selectedMovieDetails: null,
};
