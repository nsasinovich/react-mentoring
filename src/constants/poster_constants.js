export const PosterSizes = {
    LARGE: 'poster-large',
    MEDIUM: 'poster-medium',
};
