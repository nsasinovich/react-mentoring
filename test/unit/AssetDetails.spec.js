import { shallow } from 'enzyme';
import React from 'react';
import { PosterSizes } from 'src/constants/app_constants';
import Poster from 'src/components/shared/poster/poster';
import AssetDetails from 'src/components/asset_details/asset_details';

describe('<AssetDetails />', () => {
    it('should render AssetDetails and match snapshot', () => {
        const wrapper = shallow(<AssetDetails />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should contain large Poster', () => {
        const wrapper = shallow(<AssetDetails />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<Poster posterSize={PosterSizes.LARGE} />)).toBe(true);
    });
});
