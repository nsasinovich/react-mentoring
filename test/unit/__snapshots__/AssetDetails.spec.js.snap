// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssetDetails /> should contain large Poster 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AssetDetails
    asset={
      Object {
        "genres": Array [
          "genre1",
          "genre2",
        ],
        "poster_path": "poster_path",
      }
    }
    onSearchButtonClick={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="asset-details-header"
        >
          <Logo
            className="asset-details-logo"
          />
          <Button
            buttonClass="search-button button-white"
            onClick={[Function]}
          >
            Search
          </Button>
        </div>,
        <div
          className="asset-details"
        >
          <Poster
            imgUrl="poster_path"
            posterSize="poster-large"
          />
          <div
            className="asset-info"
          >
            <div
              className="wrapper"
            >
              <p
                className="title"
              />
              <p
                className="rating"
              />
            </div>
            <p
              className="genre"
            >
              genre1, genre2
            </p>
            <div
              className="wrapper"
            >
              <p
                className="year"
              >
                NaN
              </p>
              <p
                className="duration"
              />
            </div>
            <p
              className="description"
            />
          </div>
        </div>,
      ],
      "className": "asset-details-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Logo
              className="asset-details-logo"
            />,
            <Button
              buttonClass="search-button button-white"
              onClick={[Function]}
            >
              Search
            </Button>,
          ],
          "className": "asset-details-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "asset-details-logo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonClass": "search-button button-white",
              "children": "Search",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Poster
              imgUrl="poster_path"
              posterSize="poster-large"
            />,
            <div
              className="asset-info"
            >
              <div
                className="wrapper"
              >
                <p
                  className="title"
                />
                <p
                  className="rating"
                />
              </div>
              <p
                className="genre"
              >
                genre1, genre2
              </p>
              <div
                className="wrapper"
              >
                <p
                  className="year"
                >
                  NaN
                </p>
                <p
                  className="duration"
                />
              </div>
              <p
                className="description"
              />
            </div>,
          ],
          "className": "asset-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "imgUrl": "poster_path",
              "posterSize": "poster-large",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="wrapper"
                >
                  <p
                    className="title"
                  />
                  <p
                    className="rating"
                  />
                </div>,
                <p
                  className="genre"
                >
                  genre1, genre2
                </p>,
                <div
                  className="wrapper"
                >
                  <p
                    className="year"
                  >
                    NaN
                  </p>
                  <p
                    className="duration"
                  />
                </div>,
                <p
                  className="description"
                />,
              ],
              "className": "asset-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="title"
                    />,
                    <p
                      className="rating"
                    />,
                  ],
                  "className": "wrapper",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "title",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "rating",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "genre1, genre2",
                  "className": "genre",
                },
                "ref": null,
                "rendered": "genre1, genre2",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="year"
                    >
                      NaN
                    </p>,
                    <p
                      className="duration"
                    />,
                  ],
                  "className": "wrapper",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": NaN,
                      "className": "year",
                    },
                    "ref": null,
                    "rendered": NaN,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": null,
                      "className": "duration",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "description",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="asset-details-header"
          >
            <Logo
              className="asset-details-logo"
            />
            <Button
              buttonClass="search-button button-white"
              onClick={[Function]}
            >
              Search
            </Button>
          </div>,
          <div
            className="asset-details"
          >
            <Poster
              imgUrl="poster_path"
              posterSize="poster-large"
            />
            <div
              className="asset-info"
            >
              <div
                className="wrapper"
              >
                <p
                  className="title"
                />
                <p
                  className="rating"
                />
              </div>
              <p
                className="genre"
              >
                genre1, genre2
              </p>
              <div
                className="wrapper"
              >
                <p
                  className="year"
                >
                  NaN
                </p>
                <p
                  className="duration"
                />
              </div>
              <p
                className="description"
              />
            </div>
          </div>,
        ],
        "className": "asset-details-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Logo
                className="asset-details-logo"
              />,
              <Button
                buttonClass="search-button button-white"
                onClick={[Function]}
              >
                Search
              </Button>,
            ],
            "className": "asset-details-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "asset-details-logo",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "buttonClass": "search-button button-white",
                "children": "Search",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Poster
                imgUrl="poster_path"
                posterSize="poster-large"
              />,
              <div
                className="asset-info"
              >
                <div
                  className="wrapper"
                >
                  <p
                    className="title"
                  />
                  <p
                    className="rating"
                  />
                </div>
                <p
                  className="genre"
                >
                  genre1, genre2
                </p>
                <div
                  className="wrapper"
                >
                  <p
                    className="year"
                  >
                    NaN
                  </p>
                  <p
                    className="duration"
                  />
                </div>
                <p
                  className="description"
                />
              </div>,
            ],
            "className": "asset-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "imgUrl": "poster_path",
                "posterSize": "poster-large",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="wrapper"
                  >
                    <p
                      className="title"
                    />
                    <p
                      className="rating"
                    />
                  </div>,
                  <p
                    className="genre"
                  >
                    genre1, genre2
                  </p>,
                  <div
                    className="wrapper"
                  >
                    <p
                      className="year"
                    >
                      NaN
                    </p>
                    <p
                      className="duration"
                    />
                  </div>,
                  <p
                    className="description"
                  />,
                ],
                "className": "asset-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="title"
                      />,
                      <p
                        className="rating"
                      />,
                    ],
                    "className": "wrapper",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "className": "title",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "className": "rating",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "genre1, genre2",
                    "className": "genre",
                  },
                  "ref": null,
                  "rendered": "genre1, genre2",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="year"
                      >
                        NaN
                      </p>,
                      <p
                        className="duration"
                      />,
                    ],
                    "className": "wrapper",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": NaN,
                        "className": "year",
                      },
                      "ref": null,
                      "rendered": NaN,
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": null,
                        "className": "duration",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "description",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<AssetDetails /> should render AssetDetails and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AssetDetails
    asset={
      Object {
        "genres": Array [
          "genre1",
          "genre2",
        ],
        "poster_path": "poster_path",
      }
    }
    onSearchButtonClick={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="asset-details-header"
        >
          <Logo
            className="asset-details-logo"
          />
          <Button
            buttonClass="search-button button-white"
            onClick={[Function]}
          >
            Search
          </Button>
        </div>,
        <div
          className="asset-details"
        >
          <Poster
            imgUrl="poster_path"
            posterSize="poster-large"
          />
          <div
            className="asset-info"
          >
            <div
              className="wrapper"
            >
              <p
                className="title"
              />
              <p
                className="rating"
              />
            </div>
            <p
              className="genre"
            >
              genre1, genre2
            </p>
            <div
              className="wrapper"
            >
              <p
                className="year"
              >
                NaN
              </p>
              <p
                className="duration"
              />
            </div>
            <p
              className="description"
            />
          </div>
        </div>,
      ],
      "className": "asset-details-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Logo
              className="asset-details-logo"
            />,
            <Button
              buttonClass="search-button button-white"
              onClick={[Function]}
            >
              Search
            </Button>,
          ],
          "className": "asset-details-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "asset-details-logo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonClass": "search-button button-white",
              "children": "Search",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Poster
              imgUrl="poster_path"
              posterSize="poster-large"
            />,
            <div
              className="asset-info"
            >
              <div
                className="wrapper"
              >
                <p
                  className="title"
                />
                <p
                  className="rating"
                />
              </div>
              <p
                className="genre"
              >
                genre1, genre2
              </p>
              <div
                className="wrapper"
              >
                <p
                  className="year"
                >
                  NaN
                </p>
                <p
                  className="duration"
                />
              </div>
              <p
                className="description"
              />
            </div>,
          ],
          "className": "asset-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "imgUrl": "poster_path",
              "posterSize": "poster-large",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="wrapper"
                >
                  <p
                    className="title"
                  />
                  <p
                    className="rating"
                  />
                </div>,
                <p
                  className="genre"
                >
                  genre1, genre2
                </p>,
                <div
                  className="wrapper"
                >
                  <p
                    className="year"
                  >
                    NaN
                  </p>
                  <p
                    className="duration"
                  />
                </div>,
                <p
                  className="description"
                />,
              ],
              "className": "asset-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="title"
                    />,
                    <p
                      className="rating"
                    />,
                  ],
                  "className": "wrapper",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "title",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "rating",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "genre1, genre2",
                  "className": "genre",
                },
                "ref": null,
                "rendered": "genre1, genre2",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="year"
                    >
                      NaN
                    </p>,
                    <p
                      className="duration"
                    />,
                  ],
                  "className": "wrapper",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": NaN,
                      "className": "year",
                    },
                    "ref": null,
                    "rendered": NaN,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": null,
                      "className": "duration",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "description",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="asset-details-header"
          >
            <Logo
              className="asset-details-logo"
            />
            <Button
              buttonClass="search-button button-white"
              onClick={[Function]}
            >
              Search
            </Button>
          </div>,
          <div
            className="asset-details"
          >
            <Poster
              imgUrl="poster_path"
              posterSize="poster-large"
            />
            <div
              className="asset-info"
            >
              <div
                className="wrapper"
              >
                <p
                  className="title"
                />
                <p
                  className="rating"
                />
              </div>
              <p
                className="genre"
              >
                genre1, genre2
              </p>
              <div
                className="wrapper"
              >
                <p
                  className="year"
                >
                  NaN
                </p>
                <p
                  className="duration"
                />
              </div>
              <p
                className="description"
              />
            </div>
          </div>,
        ],
        "className": "asset-details-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Logo
                className="asset-details-logo"
              />,
              <Button
                buttonClass="search-button button-white"
                onClick={[Function]}
              >
                Search
              </Button>,
            ],
            "className": "asset-details-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "asset-details-logo",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "buttonClass": "search-button button-white",
                "children": "Search",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Poster
                imgUrl="poster_path"
                posterSize="poster-large"
              />,
              <div
                className="asset-info"
              >
                <div
                  className="wrapper"
                >
                  <p
                    className="title"
                  />
                  <p
                    className="rating"
                  />
                </div>
                <p
                  className="genre"
                >
                  genre1, genre2
                </p>
                <div
                  className="wrapper"
                >
                  <p
                    className="year"
                  >
                    NaN
                  </p>
                  <p
                    className="duration"
                  />
                </div>
                <p
                  className="description"
                />
              </div>,
            ],
            "className": "asset-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "imgUrl": "poster_path",
                "posterSize": "poster-large",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="wrapper"
                  >
                    <p
                      className="title"
                    />
                    <p
                      className="rating"
                    />
                  </div>,
                  <p
                    className="genre"
                  >
                    genre1, genre2
                  </p>,
                  <div
                    className="wrapper"
                  >
                    <p
                      className="year"
                    >
                      NaN
                    </p>
                    <p
                      className="duration"
                    />
                  </div>,
                  <p
                    className="description"
                  />,
                ],
                "className": "asset-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="title"
                      />,
                      <p
                        className="rating"
                      />,
                    ],
                    "className": "wrapper",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "className": "title",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "className": "rating",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "genre1, genre2",
                    "className": "genre",
                  },
                  "ref": null,
                  "rendered": "genre1, genre2",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="year"
                      >
                        NaN
                      </p>,
                      <p
                        className="duration"
                      />,
                    ],
                    "className": "wrapper",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": NaN,
                        "className": "year",
                      },
                      "ref": null,
                      "rendered": NaN,
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": null,
                        "className": "duration",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "description",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
